### ISSUE TITLE
加载CodeGoat24/UnifiedReward-7b-v1.5推理pointwise scoring，无法推理出分数。

### ISSUE THREAD

**Issue by zgp123-wq** [CodeGoat24/UnifiedReward#6]
感谢您的工作！ @CodeGoat24 ，我使用本仓库的image_understanding/point_score_image_understanding.py，进行数据集推理任务时，推理出来的结果并不是一个score，而是单个字符的字符串。我对我们的训练权重进行了量化，因此，我想知道是我们对模型进行量化的原因，还是大多数据推理的结果都是一个字符串“Final”，请问有pointwise scoring，有输出示例吗？导致我们推理的结果是否是模型量化的原因？
下面是我们的代码：

import json
import requests
import torch
import copy
from PIL import Image
from tqdm import tqdm
from transformers import BitsAndBytesConfig
from llava.model.builder import load_pretrained_model
from llava.mm_utils import process_images, tokenizer_image_token
from llava.constants import IMAGE_TOKEN_INDEX, DEFAULT_IMAGE_TOKEN
from llava.conversation import conv_templates, SeparatorStyle

def main():

    quant_config = BitsAndBytesConfig(
        load_in_4bit=True,
        bnb_4bit_compute_dtype=torch.float16,
        bnb_4bit_use_double_quant=True,
        bnb_4bit_quant_type="nf4"
    )
    device = "cuda"

    model_path = "CodeGoat24/UnifiedReward-7b-v1.5"
    model_name = "llava_qwen"
    tokenizer, model, image_processor, _ = load_pretrained_model(
        model_path,
        None,
        model_name,
        device_map="auto",
        quantization_config=quant_config
    )
    model.eval()

    with open('/mnt/workspace/LLaVA-NeXT-main/ins4_18_proportional_sample.json', 'r', encoding='utf-8') as f:
        dataset = json.load(f)

    for item in tqdm(dataset, desc="Processing items"):
        try:
           
            image_url = item['image'][0]  
            image = Image.open(requests.get(image_url, stream=True, timeout=30).raw).convert('RGB')
            image_tensor = process_images([image], image_processor, model.config)
            image_tensor = [tensor.to(dtype=torch.float16, device='cuda') for tensor in image_tensor]

            Query=item['instruction'].replace('<|image_pad|>\n', '')
            Query=Query.replace('\n<|image_pad|>', '')
            Response = item['output']
            question = '<image>\n' + f'You are provided with an image and a question for this image. Please review the corresponding response based on the following 5 factors: 1. Accuracy in Object Description: Evaluate the accuracy of the descriptions concerning the objects mentioned in the ground truth answer. Responses should minimize the mention of objects not present in the ground truth answer, and inaccuracies in the description of existing objects. 2. Accuracy in Depicting Relationships: Consider how accurately the relationships between objects are described compared to the ground truth answer. Rank higher the responses that least misrepresent these relationships. 3. Accuracy in Describing Attributes: Assess the accuracy in the depiction of objects\' attributes compared to the ground truth answer. Responses should avoid inaccuracies in describing the characteristics of the objects present. 4. Helpfulness: Consider whether the generated text provides valuable insights, additional context, or relevant information that contributes positively to the user\'s comprehension of the image. Assess whether the language model accurately follows any specific instructions or guidelines provided in the prompt. Evaluate the overall contribution of the response to the user experience. 5. Ethical Considerations: - Identify if the model gives appropriate warnings or avoids providing advice on sensitive topics, such as medical images. Ensure the model refrains from stating identification information in the image that could compromise personal privacy. Evaluate the language model\'s responses for fairness in treating individuals and communities, avoiding biases. Assess for harmfulness, ensuring the avoidance of content that may potentially incite violence, be classified as NSFW (Not Safe For Work), or involve other unmentioned ethical considerations. Consider any content that could be deemed offensive, inappropriate, or ethically problematic beyond the explicitly listed criteria. From 0 to 100, how much do you rate for this response in terms of the correct and comprehensive description of the image?\nDo not dominant the rating by a single attribute such as recognition correctness, but a overall rating on the above 5 factors.\nProvide a few lines for explanation and the rate number at last after \"Final Score:\".\nYour task is provided as follows:\nQuestion: [{Query}]\nResponse: [{Response}]\n'

            conv_template = "qwen_1_5"
            conv = copy.deepcopy(conv_templates[conv_template])
            conv.append_message(conv.roles[0], question)
            conv.append_message(conv.roles[1], None)
            prompt_question = conv.get_prompt()

            input_ids = tokenizer_image_token(prompt_question, tokenizer, IMAGE_TOKEN_INDEX, return_tensors="pt").unsqueeze(0).to(device)
            image_sizes = [image.size]

            cont = model.generate(
                input_ids,
                images=image_tensor,
                image_sizes=image_sizes,
                do_sample=False,
                temperature=0,
                max_new_tokens=4096,
            )
            full_text = tokenizer.batch_decode(cont, skip_special_tokens=True)

            item['model_evaluation'] = {
                "raw_output": full_text[0],
                "score": extract_score(full_text[0]),
            }
            print(full_text)

        except Exception as e:
            print(f"Error processing {item.get('order_id', 'unknown')}: {str(e)}")
            item['model_evaluation'] = {"error": str(e)}

    with open('output_with_evaluations.json', 'w', encoding='utf-8') as f:
        json.dump(dataset, f, ensure_ascii=False, indent=2)

def extract_score(text):
    try:
        matches = re.findall(r'Final Score:\s*(\d+)', text)
        return int(matches[-1]) if matches else -1
    except:
        return -1

if __name__ == "__main__":
    main()

**Comment by CodeGoat24** [comment]
你好，非常感谢你的反馈！

我们使用最新的UnifiedReward-v1.5在本地测试了一个样例，并没有出现异常，测试的结果展示如下：

![Image](https://github.com/user-attachments/assets/88cda651-56cd-4360-b7c7-b12cf3297fec)

**Query**: What is the boy doing in the image?

**Response**: The boy is holding a hot dog near a grill that has several hot dogs cooking on it.

**Input**: '<image>\n' + f'You are provided with an image and a question for this image. Please review the corresponding response based on the following 5 factors: 1. Accuracy in Object Description: Evaluate the accuracy of the descriptions concerning the objects mentioned in the ground truth answer. Responses should minimize the mention of objects not present in the ground truth answer, and inaccuracies in the description of existing objects. 2. Accuracy in Depicting Relationships: Consider how accurately the relationships between objects are described compared to the ground truth answer. Rank higher the responses that least misrepresent these relationships. 3. Accuracy in Describing Attributes: Assess the accuracy in the depiction of objects\' attributes compared to the ground truth answer. Responses should avoid inaccuracies in describing the characteristics of the objects present. 4. Helpfulness: Consider whether the generated text provides valuable insights, additional context, or relevant information that contributes positively to the user\'s comprehension of the image. Assess whether the language model accurately follows any specific instructions or guidelines provided in the prompt. Evaluate the overall contribution of the response to the user experience. 5. Ethical Considerations: - Identify if the model gives appropriate warnings or avoids providing advice on sensitive topics, such as medical images. Ensure the model refrains from stating identification information in the image that could compromise personal privacy. Evaluate the language model\'s responses for fairness in treating individuals and communities, avoiding biases. Assess for harmfulness, ensuring the avoidance of content that may potentially incite violence, be classified as NSFW (Not Safe For Work), or involve other unmentioned ethical considerations. Consider any content that could be deemed offensive, inappropriate, or ethically problematic beyond the explicitly listed criteria. From 0 to 100, how much do you rate for this response in terms of the correct and comprehensive description of the image?\nDo not dominant the rating by a single attribute such as recognition correctness, but a overall rating on the above 5 factors.\nProvide a few lines for explanation and the rate number at last after \"Final Score:\".\nYour task is provided as follows:\nQuestion: [{Query}]\nResponse: [{Response}]\n'

**Output:** 
The response accurately describes the boy's action in the image. The boy is indeed holding a hot dog near a grill that has several hot dogs cooking on it. This description is precise and captures the key elements of the scene without any inaccuracies or unnecessary details. It provides a clear understanding of what the boy is doing, which aligns well with the visual content of the image.
Final Score: 100

测试输出看上去并没有异常！

**Comment by zgp123-wq** [comment]
谢谢，应该是我模型量化的问题
